lib/stdlib/src/gb_sets.erl

```diff
@@ -495,6 +507,86 @@ largest_1({Key, _Smaller, nil}) ->
 largest_1({_Key, _Smaller, Larger}) ->
     largest_1(Larger).
 
+-spec lower(Element1, Set) -> none | {found, Element2} when
+    Element1 :: Element,
+    Element2 :: Element,
+    Set :: set(Element).
+lower(Key, {_, T}) ->
+    lower_1(Key, T).
+
+lower_1(_Key, nil) ->
+    none;
+lower_1(Key, {Key1, _Smaller, Larger}) when Key > Key1 ->
+    case lower_1(Key, Larger) of
+        none ->
+            {found, Key1};
+        Found ->
+            Found
+    end;
+lower_1(Key, {_Key, Smaller, _Larger}) ->
+    lower_1(Key, Smaller).
+
+-spec higher(Element1, Set) -> none | {found, Element2} when
+    Element1 :: Element,
+    Element2 :: Element,
+    Set :: set(Element).
+higher(Key, {_, T}) ->
+    higher_1(Key, T).
+
+higher_1(_Key, nil) ->
+    none;
+higher_1(Key, {Key1, Smaller, _Larger}) when Key < Key1 ->
+    case higher_1(Key, Smaller) of
+        none ->
+            {found, Key1};
+        Found ->
+            Found
+    end;
+higher_1(Key, {_Key, _Smaller, Larger}) ->
+    higher_1(Key, Larger).
+
+-spec floor(Element1, Set) -> none | {found, Element2} when
+    Element1 :: Element,
+    Element2 :: Element,
+    Set :: set(Element).
+floor(Key, {_, T}) ->
+    floor_1(Key, T).
+
+floor_1(_Key, nil) ->
+    none;
+floor_1(Key, {Key, _Smaller, _Larger}) ->
+    {found, Key};
```

Comment:
As an aside (since we decided not to bring in `floor/1` and friends at the moment), this is not `>=` as pattern matching checks term equivalence, not arithmetic equality as the module is documented as using for key comparisons. This should have been something along the lines of:

```suggestion
floor_1(Key, {Candidate, _Smaller, _Larger}) when Key == Candidate ->
    {found, Candidate};
```